### PR TITLE
[RFC] "snap run --explain" with different formating

### PR DESCRIPTION
**Pull request by mvo5**
This is #7728 with a different way to format the output (i.e. decoupling output presentation from generation a bit). Hopefully this can unblock the `snap run --explain` PR :) Feedback welcome!

### CODE REVIEW

File: cmd/snap-confine/snap-confine.c [hunk 1]
```diff
@@ -451,6 +457,8 @@ int main(int argc, char **argv)
 #if 0
 	setup_user_xdg_runtime_dir();
 #endif
+	sc_explain_header("... snap-confine");
+	sc_explain_start_section("Sandbox overview:");
```
Comment (pedronis) [inline cmd/snap-confine/snap-confine.c]: the ":" should come from a helper, not be here

File: cmd/snap-confine/snap-confine.c [hunk 2]
```diff
@@ -523,10 +531,25 @@ int main(int argc, char **argv)
 	// Now that we've dropped and regained SYS_ADMIN, we can load the
 	// seccomp profiles.
 	if (sc_apply_seccomp_profile_for_security_tag(invocation.security_tag)) {
+		sc_explain_li("Applied seccomp profile: %s",
+			   invocation.security_tag);
+                sc_explain_start_section("");
+		sc_explain("source: /var/lib/snapd/seccomp/bpf/%s.src",
```
Comment (pedronis) [inline cmd/snap-confine/snap-confine.c]: there still too much punctuation around, these need be taken care by some other kind of helper

File: cmd/snap-confine/snap-confine.c [hunk 3]
```diff
@@ -523,10 +531,25 @@ int main(int argc, char **argv)
 	// Now that we've dropped and regained SYS_ADMIN, we can load the
 	// seccomp profiles.
 	if (sc_apply_seccomp_profile_for_security_tag(invocation.security_tag)) {
+		sc_explain_li("Applied seccomp profile: %s",
+			   invocation.security_tag);
+                sc_explain_start_section("");
```
Comment (pedronis) [inline cmd/snap-confine/snap-confine.c]: still sections started with "" are a bit odd, we probably need something else to take care of this

**Comment by pedronis** [comment]
there is still too much formatting punctuation coming from input strings and not helpers. Also the section concept seems to be abused to create indentation without a intro, the corresponding code reads strangely

File: cmd/snap-confine/snap-confine.c (hunk 2 above)
Comment (mvo5) [reply cmd/snap-confine/snap-confine.c]: I added a new "sc_explain_kv()" helper to format key/value paris. However to go all the way there either needs to be a "sc_explain_li_kv()" to express listitems of key/value pairs. Or this we could add a format options struct or something else that I have not thought of yet. ideas welcome.

File: cmd/snap-confine/snap-confine.c (hunk 3 above)
Comment (mvo5) [reply cmd/snap-confine/snap-confine.c]: Yeah, I think that's right, I removed the format from "sc_start_section" now to just make it indent.

File: cmd/snap-confine/snap-confine.c [hunk 4]
```diff
@@ -458,7 +458,8 @@ int main(int argc, char **argv)
 	setup_user_xdg_runtime_dir();
 #endif
 	sc_explain_header("... snap-confine");
-	sc_explain_start_section("Sandbox overview:");
+        sc_explain("Sandbox overview");
```
Comment (zyga) [inline cmd/snap-confine/snap-confine.c]: Tabs/spaces

File: cmd/snap-confine/snap-confine.c [hunk 5]
```diff
@@ -533,18 +534,18 @@ int main(int argc, char **argv)
 	if (sc_apply_seccomp_profile_for_security_tag(invocation.security_tag)) {
 		sc_explain_li("Applied seccomp profile: %s",
 			   invocation.security_tag);
-                sc_explain_start_section("");
-		sc_explain("source: /var/lib/snapd/seccomp/bpf/%s.src",
+                sc_explain_start_section();
```
Comment (zyga) [inline cmd/snap-confine/snap-confine.c]: Tabs/spaces

File: cmd/snap-confine/snap-confine.c [hunk 6]
```diff
@@ -533,18 +534,18 @@ int main(int argc, char **argv)
...
-		sc_explain("binary: /var/lib/snapd/seccomp/bpf/%s.bin "
+		sc_explain_kv("binary", "/var/lib/snapd/seccomp/bpf/%s.bin "
 			   "(built with snap-seccomp)",
 			   invocation.security_tag);
                 sc_explain_end_section();
 		// If the process is not explicitly unconfined then load the
 		// global profile as well.
 		sc_apply_global_seccomp_profile();
 		sc_explain_li("Applied seccomp profile: global profile for all snaps");
-                sc_explain_start_section("");
```
Comment (zyga) [inline cmd/snap-confine/snap-confine.c]: Tabs/spaces

File: cmd/snap-confine/snap-confine.c (hunk 2 above)
Comment (mvo5) [reply cmd/snap-confine/snap-confine.c]: I pushed the sc_explain_li_kv() helper, feedback welcome, not great but open for ideas

**Comment by mvo5** [comment]
Closing for now, it seems this is not quite the right direction yet

**Pull request closed without merging**